R/util-CalculatePercentage.R

```diff
@@ -15,15 +15,22 @@
 #' @return A data frame containing two additional columns for the precentage value and associated string
 #' @export
 CalculatePercentage <- function(data, strCurrentCol, strTargetCol, strPercVal, strPercStrVal) {
-  if (!(strCurrentCol %in% names(data) & strTargetCol %in% names(data))) {
+  if (!(strCurrentCol %in% names(data))) {
     cli::cli_abort("Check that both {strCurrentCol} and {strTargetCol} are in data")
   }
 
-  data <- data %>%
-    dplyr::mutate(
-      {{ strPercVal }} := round(.data[[strCurrentCol]] * 100 / .data[[strTargetCol]], 1),
-      {{ strPercStrVal }} := paste0(.data[[strCurrentCol]], " / ", .data[[strTargetCol]], " (", .data[[strPercVal]], "%)")
-    )
+  if (strTargetCol %in% names(data)) {
+    data <- data %>%
+      dplyr::mutate(
+        {{ strPercVal }} := round(.data[[strCurrentCol]] * 100 / .data[[strTargetCol]], 1),
+        {{ strPercStrVal }} := paste0(.data[[strCurrentCol]], " / ", .data[[strTargetCol]], " (", .data[[strPercVal]], "%)")
+      )
+  } else {
+    data <- data %>%
+      dplyr::mutate(
+        {{ strPercStrVal }} := .data[[strCurrentCol]]
```

Comment:
The argument name str**Perc**StrVal feels a little deceiving here, and I could see why target/denominator numbers wouldn't be available soemtimes. If I'm interpreting correctly, if the denominator column isn't available you still want to be able to run this and return a new column which is just the numerator renamed?

Reply:
Yep, that would be my preference - if the target column is missing set site activation and participant enrollment to the numerator.